server/api/studio/pkg/ws/client.go

```diff
@@ -0,0 +1,285 @@
+package ws
+
+import (
+	"bytes"
+	"encoding/json"
+	"log"
+	"net/http"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/vesoft-inc/nebula-http-gateway/ccore/nebula/gateway/dao"
+	"github.com/vesoft-inc/nebula-studio/server/api/studio/pkg/auth"
+	"github.com/vesoft-inc/nebula-studio/server/api/studio/pkg/base"
+	"github.com/vesoft-inc/nebula-studio/server/api/studio/pkg/ecode"
+)
+
+const (
+	// Time allowed to write a message to the peer.
+	writeWait = 10 * time.Second
+
+	// Time allowed to read the next pong message from the peer.
+	pongWait = 60 * time.Second
+
+	// Send pings to peer with this period. Must be less than pongWait.
+	pingPeriod = (pongWait * 9) / 10
+
+	// Maximum message size allowed from peer.
+	maxMessageSize = 16 * 1024
+
+	// send buffer size
+	bufSize = 512
+
+	heartbeatRequest = "1"
+
+	heartbeatResponse = "2"
+)
+
+var (
+	newline = []byte{'\n'}
+	space   = []byte{' '}
+)
+
+var upgrader = websocket.Upgrader{
+	// ReadBufferSize:  1024,
+	// WriteBufferSize: 1024,
+	CheckOrigin: func(r *http.Request) bool {
+		return true
+	},
+	Error: func(w http.ResponseWriter, r *http.Request, status int, reason error) {
+		w.WriteHeader(status)
+		w.Write([]byte(reason.Error()))
+	},
+}
+
+// Client is a middleman between the websocket connection and the hub.
+type Client struct {
+	hub *Hub
+
+	clientInfo *auth.AuthData
+
+	// The websocket connection.
+	conn *websocket.Conn
+
+	// Buffered channel of outbound messages.
+	send chan []byte
+}
+
+func (c *Client) runNgql(msgReceived *MessageReceive) {
+	msgPost := MessagePost{}
+	msgPost.Header.MsgId = msgReceived.Header.MsgId
+	msgPost.Header.SendTime = time.Now().UnixMilli()
+	msgPost.Body.MsgType = msgReceived.Body.MsgType
+
+	gql, paramList := "", []string{}
+
+	reqGql := msgReceived.Body.Content["gql"]
+	if reqGql != nil {
+		gql, _ = reqGql.(string)
```

Comment:
I think that it's should not to be ignored. How about you ?

Reply:
`msgReceived.Body.Content["gql"]` may be a `nil` value in some case, only `paramList` is passed in request data